syntax = "proto3";

package socialnetwork;

import "google/protobuf/empty.proto";

message CreatePostRequest {
    string title = 1;
    string description = 2;
    string creator_id = 3;
    bool is_private = 4;
    repeated string tags = 5;
}

message DeletePostRequest {
    string post_id = 1;
}

message UpdatePostRequest {
    string post_id = 1;
    string title = 2;
    string description = 3;
    bool is_private = 4;
    repeated string tags = 5;
}

message GetPostRequest {
    string post_id = 1;
}

message GetPostsRequest {
    int32 page = 1;
    int32 page_size = 2;
}

message Post {
    string post_id = 1;
    string title = 2;
    string description = 3;
    string creator_id = 4;
    string created_at = 5;
    string updated_at = 6;
    bool is_private = 7;
    repeated string tags = 8;
}

message GetPostResponse {
    Post post = 1;
}

message GetPostsResponse {
    repeated Post posts = 1;
}

message CreatePostResponse{
    string post_id = 1;
}

message UpdatePostResponse{
    Post post = 1;
}

service PostService {
    rpc CreatePost(CreatePostRequest) returns (CreatePostResponse);
    rpc DeletePost(DeletePostRequest) returns (google.protobuf.Empty);

    rpc UpdatePost(UpdatePostRequest) returns (UpdatePostResponse);

    rpc GetPost(GetPostRequest) returns (GetPostResponse);

    rpc GetPosts(GetPostsRequest) returns (GetPostsResponse);
}
